### PR TITLE
Refactored RouteProgress

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

Please include a brief summary of the change and which issue is fixed (e.g., fixes [#issue](link))

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

Please describe the PR goals. Just the stuff needed to implement the fix / feature and a simple rationale. It could contain many check points if needed

### Implementation

Please include all the relevant things implemented and also rationale, clarifications / disclaimers etc. related to the approach used. It could be as self code companion comments

## Screenshots or Gifs

Please include all the media files to give some context about what's being implemented or fixed. It's not mandatory to upload screenshots or gifs, but for most of the cases it becomes really handy to get into the scope of the feature / bug being fixed and also it's REALLY useful for UI related PRs screenshot gif

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotification.kt [hunk 1]
```diff
@@ -93,28 +93,31 @@ internal class MapboxNavigationNotification : NavigationNotification {
...
+            mapboxNavigation,
+            routeProgress.currentLegProgress
+        ) { mapboxNavigation, currentLegProgress ->
+            val options = mapboxNavigation.options()
+            val legDurationRemaining = currentLegProgress.durationRemaining()
+            val timeFormatType = options.timeFormatType()
+            val arrivalTime =
+                formatTime(time, legDurationRemaining, timeFormatType, isTwentyFourHourFormat)
+            String.format(etaFormat, arrivalTime)
+        }
```
Comment (Lebedinsky) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotification.kt]: WDYT? Does it looks better or not?
```suggestion
            ifNonNull(
                mapboxNavigation?.options(), 
                routeProgress.currentLegProgress().durationRemaining()
            ) { options, legDurationRemaining  ->
                val arrivalTime = formatTime(
                    time, 
                    legDurationRemaining, 
                    options.timeFormatType(), 
                    isTwentyFourHourFormat
                )
                String.format(etaFormat, arrivalTime)
            }
```

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotification.kt [hunk 2]
```diff
@@ -280,16 +286,20 @@ internal class MapboxNavigationNotification : NavigationNotification {
...
+            currentDistanceText
+        ) { distanceFormatter, currentLegProgress, currentDistanceText ->
+            val item = currentLegProgress.currentStepProgress().distanceRemaining()
+            // The call below can return an empty spanable string. toString() will cause a NPE and ?. will not catch it.
+            val str = distanceFormatter.formatDistance(item)
+            if (str == null) {
+                val formattedDistance = str.toString()
+                currentDistanceText.toString() != formattedDistance
+            } else
+                false
```
Comment (Lebedinsky) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/MapboxNavigationNotification.kt]: Either add brackets, either single line with `else`

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/TriggerProperty.kt [hunk 3]
```diff
@@ -141,14 +141,14 @@ object TriggerProperty {
                             )
                     )
                 }
-                ifNonNull(routeProgress, routeProgress.legIndex()) { routeProgress, legIndex ->
+                ifNonNull(routeProgress, routeProgress.legIndex) { _, legIndex ->
```
Comment (Lebedinsky) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/milestone/TriggerProperty.kt]:
```suggestion
                ifNonNull(routeProgress?.legIndex) { legIndex ->
```

**Comment by RingerJK** [comment]
gonna change RouteProgress in `1.0`

**Pull request closed without merging**